geomstats/geometry/discrete_curves.py

```diff
@@ -572,6 +618,47 @@ def dist(self, point_a, point_b, **kwargs):
 
         return dist
 
+    # TODO (alebrigant): generalize from_vector_to_diagonal_matrix to k
+    # diagonals and modify tensorflow vstack and hstack.
```

Comment:
What do you mean? `from_vector_to_diagonal_matrix` works for if you input a list of size `n_samples` vectors, i.e. an array of shape `(n_samples,  dim)`, and returns a list of diagonal matrices (shape `(n_samples, dim, dim)`

Reply:
PR on this has been merged! 🎉